models/articlesModel.js

```diff
@@ -24,16 +22,24 @@ const fetchAllArticles = () => {
         votes,
         article_img_url,
         (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id) AS comment_count
-        FROM articles
-        ORDER BY created_at DESC`
-    )
-    .then((result) => {
-      // Remove body property from each article object
+        FROM articles`;
+
+  if (topic) {
+    query += ` WHERE topic = $1 ORDER BY created_at DESC`;
+    return db.query(query, [topic]).then((result) => {
+      return result.rows.map((article) => {
+        delete article.body;
```

Comment:
In your query above, you haven't selected body as a column so you shouldn't need to go through and delete them. It shouldn't be there